src/coupled_driver.cpp

```diff
@@ -329,34 +333,38 @@ void CoupledDriver::init_temperatures()
 {
   comm_.message("Initializing temperatures");
 
-  if (this->has_global_coupling_data()) {
-    auto n_global = this->get_heat_driver().n_global_elem();
+  const auto& neutronics = this->get_neutronics_driver();
+  const auto& heat = this->get_heat_driver();
+
+  if (neutronics.comm_.is_root()) {
+    auto n_global = heat.n_global_elem();
```

Comment:
Yes, I agree.  A solution could look like this:

```cpp
  auto n_global = heat.n_global_elem();
  this->comm_.sendrecv_replace(n_global, neutronics_root_, heat_root_);
  neutronics.comm_.broadcast(n_global);

  if (neutronics.comm_.is_root()) {
    temperatures_.resize({n_global});
    temperatures_prev_.resize({n_global});
  }
```

To avoid doing this again, (as you noted out below), we could make `n_global` a data member and initialize it in the construtor.  What do you think?

Reply:
Yeah, maybe that's the best thing to do

Reply:
`CoupledDriver` already has an `n_cells_` data member that's known to every process (including those in the heat comm).  I feel like it's reasonable to also have an `n_global_elem_` member that's known to every proc, too.